src/handlers/carts/carts.service.ts

```diff
@@ -1,43 +1,85 @@
-import { Inject, Injectable } from '@nestjs/common';
+import { Injectable } from '@nestjs/common';
 import { CreateCartDto } from './dto/create-cart.dto';
 import { UpdateCartDto } from './dto/update-cart.dto';
-import { UtilsService } from '../../services/utils/utils.service';
-import { ConfigModule } from '../../services/config/config.module';
-import { ConfigService } from '../../services/config/config.service';
+import { NotFoundException } from '../../exceptions/NotFoundException';
+import { CartsRepositoryService } from './carts-repository.service';
+import {
+  getUserPrimaryKey,
+  parseUserPrimaryKey,
+} from '../users/utils/user-primary-key.utils';
+import { UsersService } from '../users/users.service';
+import { BadRequestException } from '../../exceptions/BadRequestException';
+import { DATABASE_EXCEPTIONS } from '../../modules/database/constants';
+import { User, UserModel } from '../users/users.model';
+import { USER_SCHEMA_KEYS } from '../users/users.schema';
+import { getIsUserParsedPrimaryKeyValid } from '../users/utils/is-user-parsed-primary-key-valid.util';
+import { ListingResponse } from '../../shared/utils/listing-response.util';
 
 @Injectable()
 export class CartsService {
   constructor(
-    private readonly utilsService: UtilsService,
-    private readonly configService: ConfigService,
+    private cartsRepositoryService: CartsRepositoryService,
+    private usersService: UsersService,
+  ) {}
 
-    @Inject('DateService')
-    private readonly DateService,
-  ) {
-    console.log(`Current env: ${configService.getAppConfig().env}`);
-  }
+  async create(createCartDto: CreateCartDto) {
+    await this.usersService.findOne(createCartDto.userId);
 
-  average() {
-    return this.utilsService.average([1,2,3]);
-  }
+    let createdItem;
+
+    try {
+      createdItem = await this.cartsRepositoryService.create(createCartDto);
+    } catch (e) {
+      if (
+        e.__type &&
+        e.__type.includes(DATABASE_EXCEPTIONS.CONDITIONAL_CHECK_FAILED)
+      ) {
+        throw new BadRequestException('Such element already exist');
```

Comment:
I'd suggest moving usage of http-related exceptions out of service!
It's good practice on a back-end to handle all kinds of exceptions centrally (in the last middleware or exception filter in Nest).

Reply:
I read about ExceptionFilter, but I think it's not appropriate in this case. It's more comfortable to call exceptions dotly where we need them instead of having a pretty big `switch case` in one place (in ExceptionFilter). What do you think?